src/Chapter4.hs

```diff
@@ -471,11 +485,12 @@ Applicatives can be found in many applications:
 Implement the Applicative instance for our 'Secret' data type from before.
 -}
 instance Applicative (Secret e) where
-    pure :: a -> Secret e a
-    pure = error "pure Secret: Not implemented!"
+  pure :: a -> Secret e a
+  pure = Reward
 
-    (<*>) :: Secret e (a -> b) -> Secret e a -> Secret e b
-    (<*>) = error "(<*>) Secret: Not implemented!"
+  (<*>) :: Secret e (a -> b) -> Secret e a -> Secret e b
+  (<*>) (Trap y) _ = Trap y
+  (<*>) (Reward f) x = fmap f x
```

Comment:
Nice work on reusing `Functor` instance for `Secret`! :world_map:

Reply:
Thanks!